### PR TITLE
fix: metadata and well-known endpoint should not have any auth

### PR DESCRIPTION
**Pull request by emilhdiaz**
Description of changes:

FHIR and SMART-on-FHIR require that the /metadata and /.well-known/smart-configuration endpoint NOT be behind auth. 

This change shifts the registration of these routes in the express router above the authentication middleware hook, so they are evaluated before auth kicks in. I think it makes more sense to make this change here than add conditional logic in the concrete implementations of `Authorization.verifyAccessToken()` to skip access token verification. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by nguyen102** [comment]
Thanks for the PR Emil. This change makes sense, let me test the code and I'll get back to you soon.

**Comment by nguyen102** [comment]
I tested the changes in this PR to confirm that `metadata` was still reachable without Auth. I also ran Crucible tests, and those also passed. 

Thanks for thee contribution @emilhdiaz, I'll merge this PR in along with the Auth PR once it's been approved.
https://github.com/awslabs/fhir-works-on-aws-authz-rbac/pull/18

**Review by nguyen102 — APPROVED** [review]

**Comment by emilhdiaz** [comment]
Thanks @nguyen102 ! Happy to contribute. Should we make a similar change in the `SMARTHandler`? 

https://github.com/awslabs/fhir-works-on-aws-authz-smart/blob/mainline/src/smartHandler.ts#L56

**Comment by nguyen102** [comment]
Thanks for the suggestion Emil. I've created a PR for it here.
https://github.com/awslabs/fhir-works-on-aws-authz-smart/pull/6

**Pull request merged**
